client/Packages/com.beamable.server/Editor/Usam/Editor/UsamMigrator.cs

```diff
@@ -752,12 +755,27 @@ public static List<AssemblyDefinitionAsset> GetAssemblyDefinitionAssets(IDescrip
 					{
 						var assetPath = AssetDatabase.GUIDToAssetPath(id);
 						var nameQuery = $"{Path.DirectorySeparatorChar}{name}.asmdef";
+						asset = AssetDatabase.LoadAssetAtPath<AssemblyDefinitionAsset>(assetPath);
+
 						if (!assetPath.Contains(nameQuery))
 						{
-							continue;
+							//First try our best to see if the name of the assembly def is different from it's name property
+							{
+								nameQuery = $"{Path.DirectorySeparatorChar}{asset.name}.asmdef";
+								if (!assetPath.Contains(nameQuery)) // IF it's still not the asset we are looking for, reset the variable and keep looking
+								{
+									asset = null;
+									continue;
+								}
+							}
 						}
+						break;
+					}
 
-						asset = AssetDatabase.LoadAssetAtPath<AssemblyDefinitionAsset>(assetPath);
+					if (asset == null) //if the asset is still null, we don't try to add this assembly as reference, put it in a list and ask user to manually add the reference
+					{
+						errors.Add($"The assembly reference {name} could not be added, please manually add it after the migration is completed.");

```

Comment:
Here I would add more info- descriptor is missing that reference. If user has multiple services it can be useful to have that specific info.

Reply:
Oh yeah, you are right. My main point was is to specify which service is affected so if I have info about adding it manually after migration I know `which` service is missing what.